src/middleware/is-authenticated.js

```diff
@@ -0,0 +1,60 @@
+var jwt = require('jsonwebtoken');
+var db = require('../models');
+
+function isAuthenticated(model, role) {
+  return function(req, res, next) {
+    if (!req.header('x-access-token')) {
+      return res.status(401).send({
+        success: false,
+        message: 'Please login or sign up'
+      });
+    }
+
+    var token = req.header('x-access-token').split(' ')[1];
+
+    return jwt.verify(token, process.env.JWT_SECRET, function(
```

Comment:
could you put the anonymous function outside verify's parentheses, it's really hard to follow in current situation